packages/block-library/src/post-template/edit.js

```diff
@@ -186,7 +187,10 @@ export default function PostTemplateEdit( {
 			}
 			// When we preview Query Loop blocks we should prefer the current
 			// block's postType, which is passed through block context.
-			const usedPostType = previewPostType || postType;
+			const usedPostType =
```

Comment:
We need to be really careful with changes in Query Loop as it has many nuances and inherent complexity (besides the complexity we've added :) ).

This change effectively makes Query Loop kind of broken if not used in a `page`. 

To test it insert a Query Loop block in a post and try to change the `post type`. The preview and the controls are wrong.

The `previewPostType` is also important for pattern suggestions on replace.

Reply:
Thanks for highlighting, @ntsekouras. I've opened a PR that removes this new context for now, which restores the previous behaviour when changing post types: https://github.com/WordPress/gutenberg/pull/66681.

I'll work on figuring out the most appropriate post type in a follow-up PR, so the bug can be fixed on trunk as soon as possible.